### PR TITLE
[grahpql] Move SystemState into Epoch

### PR DESCRIPTION
**Pull request by stefan-mysten**
## Description 

Moves the system state summary unique fields into the Epoch struct. It removes `validatorSet` from `SystemStateSummary` as this is part of the `Epoch` anyway, and it will be retrieved correctly according to the given epoch.

Supersedes #15398.

## Test Plan 

Used existing tests.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-graphql-rpc/schema/draft_target_schema.graphql
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/schema/draft_target_schema.graphql:584]: We somehow missed to add this here.

**Review by wlmyng — COMMENTED** [review]
Could we have an e2e test that creates two epochs, makes system state changes at both, and validate that each epoch fetches the correct data?

**Comment by stefan-mysten** [comment]
> Could we have an e2e test that creates two epochs, makes system state changes at both, and validate that each epoch fetches the correct data?

Yes, let's do that! Thanks for the feedback.

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql
Comment (stefan-mysten) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:650]: I do not understand what this field is, and I have a vague recollection that @amnn wrote this description. Can go either way, as long as we're sure it's correct and to the point.

File: crates/sui-graphql-rpc/schema/draft_target_schema.graphql
Comment (stefan-mysten) [reply crates/sui-graphql-rpc/schema/draft_target_schema.graphql:599]: You got it!

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql
Comment (stefan-mysten) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:642]: In `SystemStateSummary`, the resolver's result type is `Option<StorageFund>`, so I guess that's how?

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql
Comment (wlmyng) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:651]: Oh i get it now. there is literally a (dynamic?) field on 0x3::sui::SuiSystemState called 'version', and this changes whenever the fields contained in the system state object are changed?

File: crates/sui-graphql-rpc/src/functional_group.rs
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/functional_group.rs:32]: this is a bit of a tongue twister, why not just take the doc comment for SystemStateSummary, 

/// Aspects that affect the running of the system that are managed by the validators either
/// directly, or through system transactions.

File: crates/sui-graphql-rpc/src/types/system_state_summary.rs
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/types/system_state_summary.rs:39]: not blocking this PR, but I find it strange that we use language like computation_cost  when the underlying field is safe_mode_computation_rewards, or is it just me?

**Review by wlmyng — APPROVED** [review]
looks like we found a new home for SystemState fields

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql
Comment (amnn) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:651]: > there is literally a ~(dynamic?)~ field on 0x3::sui::SuiSystemState called 'version', and this changes whenever the fields contained in the system state object are changed

That's right!

File: crates/sui-graphql-e2e-tests/tests/epoch/system_state.move
Comment (amnn) [inline crates/sui-graphql-e2e-tests/tests/epoch/system_state.move:53]: nit: whitespace.

Comment (amnn) [reply crates/sui-graphql-e2e-tests/tests/epoch/system_state.move:53]: (also in other places in this file!)

File: crates/sui-graphql-rpc/schema/current_progress_schema.graphql
Comment (amnn) [reply crates/sui-graphql-rpc/schema/current_progress_schema.graphql:640]: nit: "storage" rebates rather than "object" rebates.

File: crates/sui-graphql-rpc/schema/draft_target_schema.graphql
Comment (amnn) [inline crates/sui-graphql-rpc/schema/draft_target_schema.graphql:598]: We can also get rid of this TODO, I believe, because the fields in `EndOfEpochInfo` (on the last checkpoint in the epoch) should be indexed and used to update the relevant fields on the `epoch` table, when the epoch finishes.

File: crates/sui-graphql-rpc/schema/draft_target_schema.graphql
Comment (amnn) [inline crates/sui-graphql-rpc/schema/draft_target_schema.graphql:602]: nit: This should be an `Int` rather than a `BigInt`, I think?

File: crates/sui-graphql-rpc/src/types/system_state_summary.rs
Comment (amnn) [reply crates/sui-graphql-rpc/src/types/system_state_summary.rs:39]: It's all a matter of perspective, right? One person's reward is another person's cost.  The switch comes because in GraphQL we decided to create a reusable type to represent these running gas totals, and we are reusing it everywhere, whereas we don't have the same abstraction in the underlying representation.  So maybe there is room for that to change here, but I would do it that way around (unify it in the underlying representation) than be faithful to the names used in `sui-core`, in this case.

File: crates/sui-graphql-rpc/src/types/system_state_summary.rs
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/system_state_summary.rs:50]: As per an earlier comment, let's just make this an `Int` (just return the `u64`) because this is also a number that will not grow quickly enough to hit 2^53.

File: crates/sui-graphql-rpc/src/types/system_state_summary.rs
Comment (amnn) [inline crates/sui-graphql-rpc/src/types/system_state_summary.rs:59]: Let's make a note of this in the backlog.

**Review by amnn — APPROVED** [review]
LGTM, thank you!

**Comment by stefan-mysten** [comment]
Thanks @wlmyng and @amnn for the excellent feedback. I addressed it all and will merge once the CI is green.

**Pull request merged**
